presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -235,38 +237,25 @@ private static Partitioner createPartitioner(PartitioningHandle partitioning, in
                             taskStatsCollector,
                             toTaskProcessorBroadcastInputs(broadcastInputs)));
         }
-        else if (rddInputs.size() == 1) {
-            Entry<PlanFragmentId, JavaPairRDD<Integer, PrestoSparkRow>> input = getOnlyElement(rddInputs.entrySet());
-            PairFlatMapFunction<Iterator<Tuple2<Integer, PrestoSparkRow>>, Integer, PrestoSparkRow> taskProcessor =
-                    createTaskProcessor(
-                            executorFactoryProvider,
-                            serializedTaskDescriptor,
-                            input.getKey().toString(),
-                            taskStatsCollector,
-                            toTaskProcessorBroadcastInputs(broadcastInputs));
-            return input.getValue()
-                    .mapPartitionsToPair(taskProcessor);
```

Comment:
Then we need to have 2 different task processors implemented, one for `partitionBy` and one for `zipRdd`. This results in code duplication.

Reply:
Also I was thinking about extending `PrestoSparkZipRdd` to also accept an RDD of partitioned splits for bucketed tables. Thus I would prefer to have all exchanges implemented with `PrestoSparkZipRdd`.

Reply:
@arhimondr : 

> Then we need to have 2 different task processors implemented, one for partitionBy and one for zipRdd. This results in code duplication.

My intuition is there might be some way to refactor the task processor to avoid code duplication. I raise this as originally, the code is clear about when to use the `mapPartitions` primitive and when to use `zipPartitions` primitive. Now `zipPartitions` becomes a bit omnipotent.

But I agree let's revisit these primitives after bucketed table is implemented (even after initial production use cases 😃 )

Reply:
I have a prototype for bucketed tables.

I consolidated all RDD creation into the PrestoSparkTaskRdd, that accepts both, `PrestoSparkRow` RDD's and RDD containing splits: https://github.com/prestodb/presto/pull/14559/commits/c018b12f1031ba8c6cf88cbf8f5ffc3ccef906b4#diff-c96d6601d2e7713a8432fbd9702bdcb6R41

The `PrestoSparkTaskRdd` takes `PrestoSparkTaskProcessor`, that is also now consolidated into one: 
https://github.com/prestodb/presto/pull/14559/commits/c018b12f1031ba8c6cf88cbf8f5ffc3ccef906b4#diff-25f8e21cfd79d29a34d89b1f2758822bR32